lib/beaker/hypervisor/aws_sdk.rb

```diff
@@ -985,5 +986,16 @@ def load_fog_credentials(dot_fog = '.fog')
         :secret_key => default[:aws_secret_access_key],
       }
     end
+
+    # Adds port 8143 to host[:additional_ports]
+    # if master, database and dashboard are not on same instance
+    def test_split_install
+      @hosts.each do |host|
+        mono_roles = ['master', 'database', 'dashboard']
+        if (host[:roles] & mono_roles).size != 3 and (host[:roles] & mono_roles).any?
```

Comment:
Not sure if the test logic:

`(host[:roles] & mono_roles).size != 3`

is appropriate. The number of roles can very beyond 3, as shown at https://gist.github.com/doug-rosser/3c32a89af904d8f4dbb7bafc5103c548#file-example-aws-ec2-config-L46